src/utils/constants.js

```diff
@@ -24,19 +24,24 @@ export const READY_STATE_STAGES = {
 export const MIME_TYPES = {
   MP4: 'video/mp4',
   VND_APPLE_MPEGURL: 'application/vnd.apple.mpegurl',
+  VND_MS_SSTR: 'application/vnd.ms-sstr+xml',
 }
 
 export const MIME_TYPES_BY_EXTENSION = {
   m3u8: MIME_TYPES.VND_APPLE_MPEGURL,
   m3u: MIME_TYPES.VND_APPLE_MPEGURL,
   mp4: MIME_TYPES.MP4,
+  ism: MIME_TYPES.VND_MS_SSTR,
 }
 
 export const getExtension = url => {
   const urlWithoutParameters = url.split('?')[0] //eslint-disable-line
   const match = urlWithoutParameters.match(/\.([a-z0-9]+)$/i)
+  const matchForISM = urlWithoutParameters.match(/.+\/.+\.([a-z0-9]+)\/.+/i)
 
   return match && match[1]
```

Comment:
What about using a single regex to extract the extension from any URL?

```suggestion
  const urlWithoutParameters = url.split('?')[0] //eslint-disable-line
  const match = urlWithoutParameters.match(/(\.[A-Z0-9]+)/gi)
  const extension = match ? match.pop().replace('.', '') : ''

  return extension
```

Reply:
Nice! I've already applied this suggestion to this commit: 6e8bd8175743395eb5e7643ac82c4dd14d36e9bc